### PR TITLE
protocol: adding ftp scheme support

### LINKED ISSUE #1612: ftp support
When I try to load resource using `ftp://` protocol.

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #1612

### CODE REVIEW

**Comment by hokein** [comment]
@deepak1556 
If I remember correctly, the FTP handler have some Chromium resource dependencies.
I test it just now on Windows with `window.open('<ftp-url>')`, and it renders a blank page.

**Comment by YuriSolovyov** [comment]
Yeah, it should render folders list or something.

**Comment by deepak1556** [comment]
@hokein chromium injects some scripts and styling for the directory display, i still havnt found how it does that. But if you try to open any renderable file with this scheme it will work. So yeah this is a partial PR.

**Comment by hokein** [comment]
@deepak1556 
Yeah, Some scripts are injected to render ftp directory. [`net_resource_provider`](https://code.google.com/p/chromium/codesearch#chromium/src/chrome/common/net/net_resource_provider.cc&sq=package:chromium&type=cs&l=35&rcl=1431748312) maybe a good start point. But it relies on some resources in some chrome pak file.

**Comment by deepak1556** [comment]
@hokein its fixed now. Thanks!

File: atom/browser/atom_browser_context.cc [hunk 1]
```diff
@@ -59,6 +62,12 @@ net::URLRequestJobFactory* AtomBrowserContext::CreateURLRequestJobFactory(
       url::kFileScheme, new asar::AsarProtocolHandler(
           BrowserThread::GetBlockingPool()->GetTaskRunnerWithShutdownBehavior(
               base::SequencedWorkerPool::SKIP_ON_SHUTDOWN)));
+  auto host_resolver = Get()->url_request_context_getter()
```
Comment (zcbenz) [inline atom/browser/atom_browser_context.cc]: The `Get()->` can be removed.

File: atom/renderer/atom_render_view_observer.cc [hunk 2]
```diff
@@ -29,6 +31,11 @@ namespace atom {
 
 namespace {
 
+// A hack here:
+// Copy from net/grit/net_resources.h of chromium repository
```
Comment (zcbenz) [inline atom/renderer/atom_render_view_observer.cc]: We should copy this header in libchromiumcontent, this value can change in future.

**Comment by deepak1556** [comment]
@zcbenz fixed, depends on https://github.com/atom/libchromiumcontent/pull/119

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**